src/org/zaproxy/zap/extension/autoupdate/AutoUpdateAPI.java

```diff
@@ -54,13 +77,35 @@ public String getPrefix() {
 	@Override
 	public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
 		log.debug("handleApiAction " + name + " " + params.toString());
-		if (DOWNLOAD_LATEST_RELEASE.equals(name)) {
+		if (ACTION_DOWNLOAD_LATEST_RELEASE.equals(name)) {
 			if (this.downloadLatestRelease()) {
 				return ApiResponseElement.OK;
 			} else {
 				return ApiResponseElement.FAIL;
 			}
 
+		} else if (ACTION_INSTALL_ADDON.equals(name)) {
+		    String id = params.getString(PARAM_ID);
+		    AddOn ao = extension.getAddOn(id);
+		    if (ao == null) {
+		        throw new ApiException(Type.DOES_NOT_EXIST);
+		    } else {
+		        Vector<String> v = new Vector<String>();
+		        v.add(id);
+		        extension.installAddOns(v);
+                return ApiResponseElement.OK;
+		    }
+        } else if (ACTION_UNINSTALL_ADDON.equals(name)) {
+            String id = params.getString(PARAM_ID);
+            AddOn ao = extension.getAddOn(id);
+            if (ao == null || ! extension.getInstalledAddOns().contains(ao)) {
```

Comment:
It might fail to uninstall an add-on if the one installed is newer/older than the one available in the marketplace (they are not considered equals because of different version).